pipeline/load.py

```diff
@@ -44,65 +55,38 @@ def get_connection() -> extensions.connection:
         return None
 
 
-def get_cursor(connection: extensions.connection) -> Optional[extensions.cursor]:
+def get_cursor(connection: extensions.connection) -> extensions.cursor:
     """
-    Retrieves a the cursor for querying
-    the database from a connection.
+    Retrieves the cursor for querying the database from a connection.
     """
     try:
         return connection.cursor()
-    except:
-        logging.warning("Unable to retrieve cursor for connection")
-        return None
+    except Exception as e:
+        logging.error("Unable to retrieve cursor for connection: %s", str(e))
+        raise
 
 
-def get_id_from_country(country_name: str, cursor: extensions.cursor) -> int:
+def get_id_from_table(search_value: str, table_name: str,
+                      cursor: extensions.cursor) -> Optional[int]:
     """
-    Retrieves the country_id of a 
-    given country from the database.
+    Retrieves the id of a given value from a specified table.
     """
-    search_query = f"SELECT country_id FROM country WHERE country_name = '{
-        country_name}';"
-    cursor.execute(search_query)
-    country_id = cursor.fetchone()
-    if not country_id:
-        logging.info("Unable to find id for %s", country_name)
-        return -1
-    return country_id[0]
-
-
-def get_id_from_artist(artist_name: str, cursor: extensions.cursor) -> int:
-    """
-    Retrieves the artist id of a 
-    given artist from the database.
-    """
-    search_query = "SELECT artist_id FROM artist WHERE artist_name = %s;"
-    cursor.execute(search_query, (artist_name,))
-    artist_id = cursor.fetchone()
-    if not artist_id:
-        logging.info("Unable to find id for %s", artist_name)
-        return -1
-    return artist_id[0]
-
-
-def get_id_from_release_type(release_type: str, cursor: extensions.cursor) -> int:
-    """
-    Retrieves the release type id of a 
-    given release type.
-    """
-    search_query = "SELECT type_id FROM type WHERE type_name = %s;"
-    cursor.execute(search_query, (release_type.lower(),))
-    type_id = cursor.fetchone()
-    if not type_id:
-        logging.info("Unable to find id for %s", release_type)
-        return -1
-    return type_id[0]
+    query = f"""SELECT {table_name}_id FROM {
+        table_name} WHERE {table_name}_name = %s;"""
+    try:
+        cursor.execute(query, (search_value,))
+        result = cursor.fetchone()
+        return result[0] if result else None
+    except Exception as e:
+        logging.error("Error querying %s: %s", table_name, str(e))
+        return None
 
 
 def get_id_from_release_info(release_name: str, release_date: datetime,
-                             artist_id: int, type_id: int, cursor: extensions.cursor):
+                             artist_id: int, type_id: int,
+                             cursor: extensions.cursor) -> Optional[int]:
```

Comment:
Last week we got this feedback:
"Your return types can be dict | None, you don’t need to import optional and do Optional[dict]"
I think the same applies here

Reply:
And all the other instances in different functions